workflowai/core/domain/run_test.py

```diff
@@ -120,3 +121,86 @@ class TestRunURL:
     @patch("workflowai.env.WORKFLOWAI_APP_URL", "https://workflowai.hello")
     def test_run_url(self, run1: Run[_TestOutput]):
         assert run1.run_url == "https://workflowai.hello/agents/agent-1/runs/test-id"
+
+
+class TestFetchCompletions:
+    """Tests for the fetch_completions method of the Run class."""
+
+    # Test the successful case of fetching completions:
+    # 1. Verifies that the API client is called with the correct URL and parameters
+    # 2. Verifies that the response is properly parsed into CompletionsResponse
+    # 3. Checks that all fields (messages, response, usage) are correctly populated
+    # 4. Ensures the completion contains the expected conversation history (system, user, assistant)
+    async def test_fetch_completions_success(self, run1: Run[_TestOutput]):
```

Comment:
it would be better to add a test with a payload coming from the API, this way you would test that CompletionsResponse correctly validates the api payload.
For example here, I think that some of the completion fields are optional and so the call will fail when the payload does not contain certain fields.